picard/solver.py

```diff
@@ -136,21 +144,22 @@ def picard(X, fun='tanh', n_components=None, ortho=True, whiten=True,
     if n_components is None:
         n_components = min(n, p)
 
+    X1 = X.copy()
```

Comment:
here you make a copy everytime. See what you do:

```
    X1 = X.copy()
    if centering:
        # Center the columns (ie the variables)
        X_mean = X1.mean(axis=-1)
        X1 -= X_mean[:, np.newaxis]
    if whiten:
        # Whitening and preprocessing by PCA
        u, d, _ = linalg.svd(X1, full_matrices=False)
        del _
        K = (u / d).T[:n_components]
        del u, d
        K *= np.sqrt(p)
        X1 = np.dot(K, X1)
    else:
        # X must be casted to floats to avoid typing issues with numpy 2.0
        X1 = X1.astype('float')
```

I would do:

```
    if centering:
        X = X.copy()
        # Center the columns (ie the variables)
        X_mean = X.mean(axis=-1)
        X -= X_mean[:, np.newaxis]
    if whiten:
        # Whitening and preprocessing by PCA
        u, d, _ = linalg.svd(X, full_matrices=False)
        del _
        K = (u / d).T[:n_components]
        del u, d
        K *= np.sqrt(p)
        X = np.dot(K, X)
    else:
        # X must be casted to floats to avoid typing issues with numpy 2.0
        X = X.astype('float64')
```

Reply:
The way I code it, I do a copy everytime. But only one copy. The way you propose requires to do a 
` Y = X.copy() ` at the beginning of the calls to the internal solvers `picard_standard` or `picardo`, resulting in two copies if `whiten == True `, while there is only one copy as it is done right now.